File: 1512-number-of-good-pairs/1512-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        hash_map = {}
        ans = 0
        for num in nums:
            if num in hash_map:
                hash_map[num] += 1
                ans += hash_map[num]-1
            else:
                hash_map[num] = 1
            
        return ans

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int numIdenticalPairs(int[] nums) {
        HashMap<Integer,Integer> hashMap = new HashMap<>();
        int answer = 0;
        for (int num: nums) {
            if (hashMap.containsKey(num)) {
                int count = hashMap.get(num);
                answer += count;
                hashMap.put(num, count+1);
            } else {
                hashMap.put(num,1);
            }
        }
        return answer;
    }
}